### PR TITLE
lib: bsdlib: version 0.7.0

### PR DESCRIPTION
**Pull request by evenl**
Library version: bd0426b9c59f1bb185386867e2092ed9c2cb1118

* Major rewrite of lower transport layer to fix issue where sometimes packages where lost in high bandwidth application.
* Fixed bug in GPS socket driver which made it some times try to free the same memory twice.
* Fixed bug where TCP/IP session would sometimes hang when the transfer is done.
* Allocating more sockets than available will not return ENOBUFS instead of ENOMEM.
* GNSS socket now supports stopping a stopped GPS module to apply delete mask.
* Various GNSS documentation issues fixed.
* Add support for GPS priority setting to give GPS module priority over LTE to generate a fix.
* Added parameter checking and only return -1 on error for PDN set socket option function.
* Added support for send timeout on TCP, UDP (including secure sockets) and AT sockets.
* ai_canonname in addrinfo structure is now properly allocated and null-terminated.
* Properly disconnect the PDN if `nrf_connect()` fails.
* Ensure `bsdlib_shutdown()` works correctly.
* Added support for MSG_TRUNC, can be used with AT, GNSS, TCP, UDP sockets.

Corresponding nrf PR: https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/2157

Signed-off-by: Even Falch-Larsen <even.falch-larsen@nordicsemi.no>

### CODE REVIEW

File: bsdlib/doc/CHANGELOG.rst [hunk 1]
```diff
@@ -5,6 +5,23 @@ Changelog
...
+* Allocating more sockets than available will not return ENOBUF instead of ENOMEM.
+* GNSS socket no support stopping a stopped GPS module to apply delete mask.
+* Various GNSS documentation issues fixed.
+* Add support for GPS priority setting to give GPS module priority over LTE to generate a fix
+* Added parameter checking and only return -1 on error for PDN set socket option function.
+* Added support for send timeout on TCP, UDP (including secure sockets) and AT sockets.
+* ai_canonname in addrinfo structure is now properly allocated and zero terminated.
+* Properly disconnect the PDN if `nrf_connect()` failes.
+* Ensure `bsdlib_shutdown()` works correctly.
+* Added support for MSG_TRUNK, can be used with AT, GNSS, TCP, UDP sockets.
```
Comment (rlubos) [inline bsdlib/doc/CHANGELOG.rst:23]: s/MSG_TRUNK/MSG_TRUNC

File: bsdlib/doc/CHANGELOG.rst [hunk 2]
```diff
@@ -5,6 +5,23 @@ Changelog
...
+* Fixed bug in GPS socket driver which made it some times try to free the same memory twice.
+* Fixed bug where TCP/IP session would sometimes hang when the transfer is done.
+* Allocating more sockets than available will not return ENOBUF instead of ENOMEM.
+* GNSS socket no support stopping a stopped GPS module to apply delete mask.
+* Various GNSS documentation issues fixed.
+* Add support for GPS priority setting to give GPS module priority over LTE to generate a fix
+* Added parameter checking and only return -1 on error for PDN set socket option function.
+* Added support for send timeout on TCP, UDP (including secure sockets) and AT sockets.
+* ai_canonname in addrinfo structure is now properly allocated and zero terminated.
+* Properly disconnect the PDN if `nrf_connect()` failes.
```
Comment (rlubos) [inline bsdlib/doc/CHANGELOG.rst:21]: s/failes/fails

File: bsdlib/doc/CHANGELOG.rst [hunk 3]
```diff
@@ -5,6 +5,23 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+bsdlib 0.7.0
+************
+
+* Major rewrite of lower transport layer to fix issue where sometimes packages where lost in high bandwidth application.
+* Fixed bug in GPS socket driver which made it some times try to free the same memory twice.
+* Fixed bug where TCP/IP session would sometimes hang when the transfer is done.
+* Allocating more sockets than available will not return ENOBUF instead of ENOMEM.
```
Comment (rlubos) [inline bsdlib/doc/CHANGELOG.rst:14]: s/ENOBUF/ENOBUFS

File: bsdlib/doc/CHANGELOG.rst [hunk 4]
```diff
@@ -5,6 +5,23 @@ Changelog
...
+* GNSS socket no support stopping a stopped GPS module to apply delete mask.
```
Comment (rlubos) [inline bsdlib/doc/CHANGELOG.rst:15]: I'm confused with the first part, shouldn't it be "GNSS socket no**w** soupports"?

File: bsdlib/doc/CHANGELOG.rst (hunk 1 above)
Comment (evenl) [reply bsdlib/doc/CHANGELOG.rst]: fixed

File: bsdlib/doc/CHANGELOG.rst [hunk 5]
```diff
@@ -5,6 +5,23 @@ Changelog
...
+* Various GNSS documentation issues fixed.
+* Add support for GPS priority setting to give GPS module priority over LTE to generate a fix
+* Added parameter checking and only return -1 on error for PDN set socket option function.
+* Added support for send timeout on TCP, UDP (including secure sockets) and AT sockets.
+* ai_canonname in addrinfo structure is now properly allocated and zero terminated.
```
[the remark this answers is not in the document]
Comment (evenl) [reply bsdlib/doc/CHANGELOG.rst:20]: hehe, yes, more common, fixing

File: bsdlib/doc/CHANGELOG.rst (hunk 3 above)
Comment (evenl) [reply bsdlib/doc/CHANGELOG.rst:14]: ok

File: bsdlib/doc/CHANGELOG.rst (hunk 4 above)
Comment (evenl) [reply bsdlib/doc/CHANGELOG.rst]: yes, fixed to "GNSS socket now supports stopping a stopped GPS module to apply delete mask."

**Comment by evenl** [comment]
Do I need to care about the line lenght in the CHANGELOG.rst file? I see that the build job is not happy, but the file is full or lines over 80 chars.

**Comment by b-gent** [comment]
> Do I need to care about the line lenght in the CHANGELOG.rst file? I see that the build job is not happy, but the file is full or lines over 80 chars.

RST files are not affected by the 80 chars limit, instead we have a 'one sentence - one line' rule in them.

**Comment by rlubos** [comment]
@evenl The CI complains about the commit message, not the content of the changelog file.

**Comment by evenl** [comment]
Doh! Sorry about that, fixed.

image

File: bsdlib/doc/CHANGELOG.rst [hunk 6]
```diff
@@ -5,6 +5,23 @@ Changelog
...
+bsdlib 0.7.0
+************
+
+* Major rewrite of the lower transport layer to fix an issue where packages where lost in a high bandwidth application.
+* Added support for GPS priority setting to give the GPS module priority over LTE to generate a fix.
+* Added parameter checking and only return -1 on error for the PDN set socket option function.
+* Added support for send timeout on TCP, UDP (including secure sockets), and AT sockets.
+* Added support for MSG_TRUNC on AT, GNSS, TCP, and UDP sockets.
+* Allocating more sockets than available will now return ENOBUFS instead of ENOMEM.
+* GNSS socket now supports stopping a stopped GPS module to apply a delete mask.
```
Comment (b-gent) [inline bsdlib/doc/CHANGELOG.rst:17]: "stopping a stopped GPS module" -> is this how it's supposed to be?

**Comment by b-gent** [comment]
@evenl  can you check my commit please. I changed the order of the points quite a bit , leaving the 'major rewrite' first (it seems to me like a highlight of the release) and after that I grouped the points into added, changed, and fixed. 

Also, leaving one question in the comment.

File: bsdlib/doc/CHANGELOG.rst (hunk 6 above)
Comment (evenl) [reply bsdlib/doc/CHANGELOG.rst:17]: Yes, a bit strange sentence, but it has to do with the state machine in the GNSS driver. Before you where not allowed to try to transition to STOPPED state if you where already in STOPPED state. The reason you want to to do that is to apply a delete mask without having to transition to STARTED and then back to STOPPED mode if you only want to apply this delete mask.

**Comment by evenl** [comment]
Squashed and pusshed the changelog updates

**Comment by b-gent** [comment]
Looking good.

**Comment by evenl** [comment]
@eibe85 Just updated the PR with the fix to the bsdlib_shutdown() -> bsdlib_init() sequence. Can you have a look and approve?

**Comment by lemrey** [comment]
nit: You don't need the `lib: ` prefix in the commit message.

File: bsdlib/doc/CHANGELOG.rst [hunk 7]
```diff
@@ -5,6 +5,23 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+bsdlib 0.7.0
+************
+
+* Major rewrite of the lower transport layer to fix an issue where packages where lost in a high bandwidth application.
```
Comment (kacperradoszewski) [inline bsdlib/doc/CHANGELOG.rst:11]: small typo: 

> [...] fix an issue where packages **where** lost [...]

Comment (b-gent) [reply bsdlib/doc/CHANGELOG.rst:11]: This was corrected in my commit, must have been overwritten?
https://github.com/NordicPlayground/nrfxlib/commit/c50ae953ae19a61daca75a44fc4821ae9f325e83

**Comment by b-gent** [comment]
@evenl  I pushed my changes again

**Pull request merged**